src/main/java/software/amazon/lambda/snapstart/ByteCodeIntrospector.java

```diff
@@ -80,6 +80,18 @@ boolean implementsLambdaInterface(XClass xClass) {
         return false;
     }
 
+    /**
+     * This returns true only if this class is used as a field in a Lambda handler class
+     */
+    boolean isLambdaHandlerField(XClass xClass) {
+        for (String fieldType : CacheLambdaHandlerFields.fieldsToVisit) {
```

Comment:
I would add a method in `CacheLambdaHandlerFields` class, so you don't access the variable `fieldsToVisit` directly.

Reply:
We could even add a more useful method even, that you pass an `xClass` and it returns if it's the class of a field in the cache or not (so this `for` loop is inside that method instead of being here).

Reply:
I second this, it's worth considering to add a new database type.